dev/com.ibm.ws.repository.parsers/test/com/ibm/ws/repository/parsers/internal/CreateJsonRepositoryFilesTest.java

```diff
@@ -0,0 +1,148 @@
+
+        // tidy up just in case tests next tests fails to produce one and assumes this is its output
+        createdFile.delete();
+    }
+
+    /**
+     * Test that the JSON created for the same asset dosn't change with other product or parser changes.
+     * The metadata file used is explicitly specified. We will only test the specified metadata option
+     * for feature, the rest will only test co-located as if this works they should all do so.
+     *
+     * @throws IOException
+     */
+    @Test
+    public void testJspJsonWithSpecifiedMetadata() throws IOException {
+        String name = "com.ibm.websphere.appserver.jsp-2.3.esa";
+
+        File assetFile = new File(baseDir, "resources/esa/esaAssetOnly/" + name);
+        File metadataFile = new File(baseDir, "resources/esa/esaMetadataOnly/" + name + ".metadata.zip");
+        String assetType = ResourceType.FEATURE.toString();
+        String outputLocation = new File(baseDir, "/build/json").getAbsolutePath();
+        runAsIfCalledFromXml(assetFile, metadataFile, assetType, outputLocation);
+
+        // compare the output json with the reference json
+        File createdFile = new File(baseDir, "/build/json/" + name + ".json");
+        File referenceFile = new File(baseDir, "/resources/reference/" + name + ".json");
+        assertThat(readJsonObject(createdFile), matchesJsonInAnyOrder(readJsonObject(referenceFile)));
+
+        // tidy up just in case tests next tests fails to produce one and assumes this is its output
+        createdFile.delete();
+    }
+
+    /**
+     * Test that the JSON created for the same asset dosn't change with other product or parser changes.
+     * The metadata file used is not specified but there is one co-located with the esa.
+     *
+     * @throws IOException
+     */
+    @Test
+
+    public void testEsaJson() throws IOException {
+        testSomeJson("com.ibm.websphere.appserver.anno-1.0.esa", "esa", ResourceType.FEATURE.toString());
+    }
+
+    /**
+     * Test creating JSON for a parameterised call
+     *
+     * @throws IOException
+     */
+    private void testSomeJson(String name, String srcDir, String assetType) throws IOException {
+
+        File assetFile = new File(baseDir, "/resources/" + srcDir + "/" + name);
+        File metadataFile = new File("null"); // <<<<<< co-located metadata file
+        String outputLocation = new File(baseDir, "/build/json").getAbsolutePath();
+        runAsIfCalledFromXml(assetFile, metadataFile, assetType, outputLocation);
+
+        // compare the output json with the reference json
+        File createdFile = new File(baseDir, "/build/json/" + name + ".json");
+        File referenceFile = new File(baseDir, "/resources/reference/" + name + ".json");
+        assertThat(readJsonObject(createdFile), matchesJsonInAnyOrder(readJsonObject(referenceFile)));
+
+        // tidy up just in case tests next tests fails to produce one and assumes this is its output
+        createdFile.delete();
```

Comment:
Cleanup should be done in a try-finally or in an `@After` method. As it is, the cleanup won't happen if the test fails.